Packs/MongoDB/Integrations/MongoDB/MongoDB.py

```diff
@@ -180,13 +180,23 @@ def convert_str_to_datetime(entries: dict) -> dict:
     """
     # regex for finding timestamp in string
     regex_for_timestamp = re.compile(r'\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d(?:\.\d+)?Z?')
-    for key, value in entries.items():
-        for match in regex_for_timestamp.findall(str(value)):
+    time_format = '%Y-%m-%dT%H:%M:%S.000Z'
+    if isinstance(entries, str):
+        matches = regex_for_timestamp.findall(entries)
+        if len(matches) == 1:
             try:
-                entries[key] = datetime.strptime(match, '%Y-%m-%dT%H:%M:%S.000Z')
-            except ValueError as e:
-                raise DemistoException(str(e))
-
+                return datetime.strptime(matches[0], time_format)
+            except ValueError:
+                # If could not parse the date, it may not a date. Return entries
+                return entries
+        # If no match or more than 1, return itself
```

Comment:
why ?

Reply:
because if there are two matches for the regex, which one would you pick as a timestamp?
if there are two valid timestamps in the string, it's not a timestamp (non-MongoDB valid)